python/thunder/imgprocessing/register.py

```diff
@@ -1,252 +1,316 @@
+        Images object based on the key. Because transformations are small,
+        we broadcast the transformations rather than using a join.
+
+        See also
+        --------
+        Register : construct registration algorithms
+        """
+
+        from thunder.rdds.images import Images
+
+        # broadcast the transformations
+        transformations_bc = images.rdd.context.broadcast(self.transformations)
+
+        # apply the transformations
+        newrdd = images.rdd.map(lambda (k, im): (k, transformations_bc.value[k].apply(im)))
+        return Images(newrdd).__finalize__(images)
+
+    def save(self, file):
+        """
+        Serialize registration model to a file as text using JSON.
+
+        Format is a dictionary, with keys 'regmethod' and 'transtype' specifying
+        the registration method used and the transformation type (as strings),
+        and 'transformations' containing the transformations. The exact format of the transformations
+        will vary by type, but will always be a dictionary, with keys indexing into an Images object,
+        and values containing the transformation parameters.
+
+        Parameters
+        ----------
+        file : filename or file handle
+            The file to write to
+
+        """
+
+        import json
+
+        if hasattr(file, 'write'):
+            f = file
+        else:
+            f = open(file, 'w')
+        output = json.dumps(self, default=lambda v: v.__dict__)
+        f.write(output)
+        f.close()
+
+    @staticmethod
+    def load(file):
+        """
+        Deserialize registration model from a file containing JSON.
+
+        Assumes a JSON formatted registration model, with keys 'regmethod' and 'transclass' specifying
+        the registration method used and the transformation type as strings, and 'transformations'
+        containing the transformations. The format of the transformations will depend on the type,
+        but it should be a dictionary of key value pairs, where the keys are keys of the target
+        Images object, and the values are arguments for reconstructing each transformation object.
+
+        Parameters
+        ----------
+        file : str
+            Name of a file to read from
+
+        Returns
+        -------
+        model : RegisterModel
+            Instance of a registration model
+        """
+
+        import json
+        import importlib
+
+        f = open(file, 'r')
+        input = json.loads(f.read())
+
+        # import the appropriate transformation class
+        regmethod = str(input['regmethod'])
+        classname = str(input['transclass'])
+        transclass = getattr(importlib.import_module('thunder.imgprocessing.transformation'), classname)
+
+        # instantiate the transformations and construct the model
+        transformations = {int(k): transclass(**v) for k, v in input['transformations'].iteritems()}
```

Comment:
This would require that each `Transform` object can take its `__dict__` as an argument, right? You could overload the default `JSONEncoder` in `Transform` to use `getParams()` and use something like `setParams()` to recreate the object.